draft-bbs-signatures.md

```diff
@@ -680,38 +680,56 @@ Procedure:
 
 # Utility Operations
 
-## CreateGenerators
+## Generator point computation
 
-The CreateGenerators operation defines how to create a set of generators that form a part of the public parameters used by the BBS Signature scheme to accomplish operations such as Sign, Verify, ProofGen and ProofVerify.
+The `create_generators` operation defines how to create a set of generators that form a part of the public parameters used by the BBS Signature scheme to accomplish operations such as Sign, Verify, ProofGen and ProofVerify. It takes one input, the number of generator points to create, which is determined in part by the number of selectively-revealed signed messages.
+
+As an optimization, implementations MAY cache the result of `create_generators` for a specific `generator_seed` (determined by the ciphersuite) and `count`. The values `c` and `v` MAY also be cached in order to efficiently extend a existing list of generator points.
 
 ```
-generators = CreateGenerators(dst, message_generator_seed, length);
+generators = create_generators(count)
 
 Inputs:
 
-- dst (REQUIRED), octet string. Domain Separation Tag.
-- message_generator_seed (REQUIRED), octet string.
-- length (REQUIRED), unsigned integer. Number of generators to create from the seed and dst.
+- count (REQUIRED), unsigned integer. Number of generator points to create.
+
+Parameters:
+
+- generator_seed, octet string. A seed value selected by the ciphersuite.
+- hash_to_curve_g1. A hash-to-curve suite selected by the ciphersuite.
+
+Definitions:
+
+- expand_message is the corresponding operation associated with the ciphersuite's hash_to_curve_g1 parameter.
+- seed_dst is the octet string representing "BBS-SIG-GENERATOR-SEED" in the ASCII character encoding.
+- hash_output_bytes is the integer given by ceil((log2(r) + k) / 8).
+- generator_dst is the octet string representing "BBS-SIG-GENERATOR-DST" in the ASCII character encoding.
 
 Outputs:
 
 - generators, an array of generators.
 
 Procedure:
 
-1. h = xof(seed)
+1. n = 1
+
+2. v = expand_message(generator_seed, seed_dst, hash_output_bytes)
```

Comment:
That could be a little harder to implement, because generator_seed doesn't have a bounded length, so the implementer might need to allocate a separate buffer to create the input for expand_message.

Reply:
Good point. Maybe then calculating a first `v` as `v = expand_message(generator_seed, seed_dst, hash_output_bytes)` and then again as `v = expand_message(v || I2OSP(0, 4), seed_dst, hash_output_bytes)` (this second `v` will be the "seed" for the first generator)?? I do realize this adds complexity, but it will make it easier to move to a cut-off or prefix-free like construction if we decide to do so.